// @centreon/react-components/lib/ExtensionDeletePopup/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _popup = _interopRequireDefault(require("../Popup/popup.scss"));

var _Popup = _interopRequireDefault(require("../Popup"));

var _Title = _interopRequireDefault(require("../Title"));

var _MessageInfo = _interopRequireDefault(require("../Message/MessageInfo"));

var _ButtonRegular = _interopRequireDefault(require("../Button/ButtonRegular"));

var _IconClose = _interopRequireDefault(require("../Icon/IconClose"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _typeof(obj) { if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

var ExtensionDeletePopup =
/*#__PURE__*/
function (_React$Component) {
  _inherits(ExtensionDeletePopup, _React$Component);

  function ExtensionDeletePopup() {
    _classCallCheck(this, ExtensionDeletePopup);

    return _possibleConstructorReturn(this, _getPrototypeOf(ExtensionDeletePopup).apply(this, arguments));
  }

  _createClass(ExtensionDeletePopup, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          deletingEntity = _this$props.deletingEntity,
          onConfirm = _this$props.onConfirm,
          onCancel = _this$props.onCancel;
      return _react.default.createElement(_Popup.default, {
        popupType: "small"
      }, _react.default.createElement("div", {
        className: (0, _classnames.default)(_popup.default["popup-header"])
      }, _react.default.createElement(_Title.default, {
        label: deletingEntity.description
      })), _react.default.createElement("div", {
        className: (0, _classnames.default)(_popup.default["popup-body"])
      }, _react.default.createElement(_MessageInfo.default, {
        messageInfo: "red",
        text: "Do you want to delete this extension? This action will remove all associated data."
      })), _react.default.createElement("div", {
        className: (0, _classnames.default)(_popup.default["popup-footer"])
      }, _react.default.createElement("div", {
        className: (0, _classnames.default)(_popup.default["container__row"])
      }, _react.default.createElement("div", {
        className: (0, _classnames.default)(_popup.default["container__col-xs-6"])
      }, _react.default.createElement(_ButtonRegular.default, {
        label: "Delete",
        buttonType: "regular",
        color: "red",
        onClick: function onClick(e) {
          e.preventDefault();
          e.stopPropagation();
          onConfirm(deletingEntity.id, deletingEntity.type);
        }
      })), _react.default.createElement("div", {
        className: (0, _classnames.default)(_popup.default["container__col-xs-6"], ["text-left"])
      }, _react.default.createElement(_ButtonRegular.default, {
        label: "Cancel",
        buttonType: "regular",
        color: "gray",
        onClick: function onClick(e) {
          e.preventDefault();
          e.stopPropagation();
          onCancel();
        }
      })))), _react.default.createElement(_IconClose.default, {
        iconPosition: "icon-close-position-middle",
        iconType: "middle",
        onClick: function onClick(e) {
          e.preventDefault();
          e.stopPropagation();
          onCancel();
        }
      }));
    }
  }]);

  return ExtensionDeletePopup;
}(_react.default.Component);

var _default = ExtensionDeletePopup;
exports.default = _default;
