// @centreon/react-components/lib/DynamicComponentLoader/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireWildcard(require("react"));

var _reactRedux = require("react-redux");

var _Axios = _interopRequireDefault(require("../Axios"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; return newObj; } }

function _typeof(obj) { if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var DynamicComponentLoader =
/*#__PURE__*/
function (_Component) {
  _inherits(DynamicComponentLoader, _Component);

  function DynamicComponentLoader() {
    var _getPrototypeOf2;

    var _this;

    _classCallCheck(this, DynamicComponentLoader);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _possibleConstructorReturn(this, (_getPrototypeOf2 = _getPrototypeOf(DynamicComponentLoader)).call.apply(_getPrototypeOf2, [this].concat(args)));

    _defineProperty(_assertThisInitialized(_this), "state", {
      componentLoaded: false,
      componentExists: false
    });

    _defineProperty(_assertThisInitialized(_this), "componentWillReceiveProps", function (nextProps) {
      var componentName = nextProps.componentName;

      if (componentName != _this.props.componentName) {
        document.removeEventListener("component".concat(_this.props.componentName, "Loaded"), _this.setComponentLoaded);
        document.addEventListener("component".concat(componentName, "Loaded"), _this.setComponentLoaded);

        _this.checkFileExists();
      }
    });

    _defineProperty(_assertThisInitialized(_this), "checkFileExists", function () {
      var _this$props = _this.props,
          componentUrl = _this$props.componentUrl,
          xhr = _this$props.xhr;
      xhr({
        requestType: "GET",
        url: componentUrl,
        check: true
      }).then(function () {
        _this.setState({
          componentExists: true
        });
      }).catch(function (err) {
        _this.setState({
          componentExists: false
        });
      });
    });

    _defineProperty(_assertThisInitialized(_this), "componentWillMount", function () {
      if (_this.props.componentName) {
        document.addEventListener("component".concat(_this.props.componentName, "Loaded"), _this.setComponentLoaded);

        _this.checkFileExists();
      }
    });

    _defineProperty(_assertThisInitialized(_this), "setComponentLoaded", function () {
      _this.setState({
        componentLoaded: true
      });
    });

    _defineProperty(_assertThisInitialized(_this), "componentWillUnmount", function () {
      var componentName = _this.props.componentName;
      document.removeEventListener("component".concat(componentName, "Loaded"), _this.setComponentLoaded);
    });

    return _this;
  }

  _createClass(DynamicComponentLoader, [{
    key: "render",
    value: function render() {
      var _this$state = this.state,
          componentLoaded = _this$state.componentLoaded,
          componentExists = _this$state.componentExists;
      var componentUrl = this.props.componentUrl;
      return _react.default.createElement(_react.default.Fragment, null, componentExists ? _react.default.createElement("iframe", {
        src: componentUrl,
        style: _defineProperty({
          width: 0,
          height: 0,
          border: "0"
        }, "border", "none")
      }) : null);
    }
  }]);

  return DynamicComponentLoader;
}(_react.Component);

var mapDispatchToProps = function mapDispatchToProps(dispatch) {
  return {
    xhr: function xhr(data) {
      var requestType = data.requestType;
      return (0, _Axios.default)(data, dispatch, requestType);
    }
  };
};

var _default = (0, _reactRedux.connect)(null, mapDispatchToProps)(DynamicComponentLoader);

exports.default = _default;
