// @centreon/react-components/lib/InputField/InputFieldSelectCustom/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireWildcard(require("react"));

var _reactDom = require("react-dom");

var _classnames = _interopRequireDefault(require("classnames"));

var _inputField = _interopRequireDefault(require("./input-field.scss"));

var _ScrollBar = _interopRequireDefault(require("../../ScrollBar"));

var _CustomIconWithText = _interopRequireDefault(require("../../Custom/CustomIconWithText"));

var _IconToggleSubmenu = _interopRequireDefault(require("../../Icon/IconToggleSubmenu"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; return newObj; } }

function _typeof(obj) { if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var InputFieldSelectCustom =
/*#__PURE__*/
function (_Component) {
  _inherits(InputFieldSelectCustom, _Component);

  function InputFieldSelectCustom() {
    var _getPrototypeOf2;

    var _this;

    _classCallCheck(this, InputFieldSelectCustom);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _possibleConstructorReturn(this, (_getPrototypeOf2 = _getPrototypeOf(InputFieldSelectCustom)).call.apply(_getPrototypeOf2, [this].concat(args)));

    _defineProperty(_assertThisInitialized(_this), "state", {
      active: false,
      allOptions: [],
      options: [],
      selected: {}
    });

    _defineProperty(_assertThisInitialized(_this), "toggleSelect", function () {
      var active = _this.state.active;

      _this.setState({
        active: !active
      });
    });

    _defineProperty(_assertThisInitialized(_this), "componentWillMount", function () {
      var _this$props = _this.props,
          value = _this$props.value,
          options = _this$props.options;

      for (var i = 0; i < options.length; i++) {
        if (options[i].id == value) {
          _this.setState({
            selected: options[i]
          });
        }
      }

      if (options) {
        _this.setState({
          options: options,
          allOptions: options
        });
      }
    });

    _defineProperty(_assertThisInitialized(_this), "componentWillReceiveProps", function (nextProps) {
      var value = nextProps.value,
          options = nextProps.options;

      for (var i = 0; i < options.length; i++) {
        if (options[i].id == value) {
          _this.setState({
            selected: options[i]
          });
        }
      }

      if (options) {
        _this.setState({
          options: options,
          allOptions: options
        });
      }
    });

    _defineProperty(_assertThisInitialized(_this), "searchTextChanged", function (e) {
      var searchString = e.target.value;
      var allOptions = _this.state.allOptions;

      _this.setState({
        options: allOptions.filter(function (option) {
          return String(option.name).toLowerCase().indexOf(String(searchString).toLowerCase()) > -1;
        })
      });
    });

    _defineProperty(_assertThisInitialized(_this), "optionChecked", function (option) {
      var onChange = _this.props.onChange;

      _this.setState({
        selected: option,
        active: false
      }, function () {
        if (onChange) {
          onChange(option.id);
        }
      });
    });

    _defineProperty(_assertThisInitialized(_this), "handleClickOutside", function (e) {
      if (!_this.select || _this.select.contains(e.target)) {
        return;
      }

      _this.setState({
        active: false
      });
    });

    _defineProperty(_assertThisInitialized(_this), "focusInput", function (component) {
      var allOptions = _this.state.allOptions;

      if (component) {
        _this.setState({
          options: allOptions
        });

        (0, _reactDom.findDOMNode)(component).focus();
      }
    });

    return _this;
  }

  _createClass(InputFieldSelectCustom, [{
    key: "UNSAFE_componentWillMount",
    value: function UNSAFE_componentWillMount() {
      window.addEventListener("mousedown", this.handleClickOutside, false);
    }
  }, {
    key: "componentWillUnmount",
    value: function componentWillUnmount() {
      window.removeEventListener("mousedown", this.handleClickOutside, false);
    }
  }, {
    key: "render",
    value: function render() {
      var _this2 = this;

      var _this$state = this.state,
          active = _this$state.active,
          selected = _this$state.selected,
          options = _this$state.options;
      var _this$props2 = this.props,
          size = _this$props2.size,
          label = _this$props2.label,
          error = _this$props2.error,
          icons = _this$props2.icons,
          domainPath = _this$props2.domainPath;
      return _react.default.createElement("div", {
        className: (0, _classnames.default)(_inputField.default["input-select"], _inputField.default[size ? size : ""], _inputField.default[active ? "active" : ""], error ? _inputField.default["has-danger"] : ""),
        ref: function ref(select) {
          return _this2.select = select;
        }
      }, _react.default.createElement("div", {
        className: (0, _classnames.default)(_inputField.default["input-select-wrap"])
      }, active ? _react.default.createElement("input", {
        ref: this.focusInput,
        onChange: this.searchTextChanged,
        className: (0, _classnames.default)(_inputField.default["input-select-input"]),
        type: "text",
        placeholder: "Search"
      }) : _react.default.createElement("span", {
        className: (0, _classnames.default)(_inputField.default["input-select-field"]),
        onClick: this.toggleSelect.bind(this)
      }, selected.name), _react.default.createElement(_IconToggleSubmenu.default, {
        iconPosition: "icons-toggle-position-multiselect",
        iconType: "arrow",
        onClick: this.toggleSelect.bind(this)
      })), active ? _react.default.createElement("div", {
        className: (0, _classnames.default)(_inputField.default["input-select-dropdown"])
      }, options ? options.map(function (option) {
        return _react.default.createElement(_react.default.Fragment, null, icons ? _react.default.createElement(_CustomIconWithText.default, {
          label: option.name,
          onClick: _this2.optionChecked.bind(_this2, option),
          image: "".concat(domainPath, "/").concat(option.preview)
        }) : _react.default.createElement("span", {
          onClick: _this2.optionChecked.bind(_this2, option),
          className: (0, _classnames.default)(_inputField.default["input-select-label"])
        }, option.name));
      }) : null) : null, error ? _react.default.createElement("div", {
        className: (0, _classnames.default)(_inputField.default["form-error"])
      }, error) : null);
    }
  }]);

  return InputFieldSelectCustom;
}(_react.Component);

var _default = InputFieldSelectCustom;
exports.default = _default;
